api/user.go

```diff
@@ -23,6 +23,7 @@ type User struct {
 	Name          string          `json:"name"`
 	ProviderNames []string        `json:"providerNames,omitempty"`
 	PublicKeys    []UserPublicKey `json:"publicKeys,omitempty"`
+	SSHUsername   string          `json:"sshUsername"`
```

Comment:
There will be other cases where a username will be needed so it doesn't make sense to create a column specific to ssh. It'd be better to create a column called `username` or `login`

Reply:
I agree, I don't really like the name `sshUsername`, and I would like to find something better.

If we call it `username` then I think we'd want it to be the users primary username throughout the system. The only problem I see with that is then we're forced to restrict these to the same rules of Linux usernames, which may be very restrictive.

Let's keep thinking about the right term to use here. Some other ideas:
* `linuxUsername`, `unixUsername`, or `posixUsername` - more specific, but maybe we'll want to use this for windows SSH or remote connections as well?
* (i'm out of ideas, but I think we can find something better)

Reply:
I don't think we need to enforce Linux username rules in this field. We can set some sensible rules and transform them to Linux username if necessary. Same with Windows.

[`useradd(8)`](https://man7.org/linux/man-pages/man8/useradd.8.html) has reasonable rules:

```
       Usernames must start with a lower case letter or an underscore,
       followed by lower case letters, digits, underscores, or dashes.
       They can end with a dollar sign. In regular expression terms:
       [a-z_][a-z0-9_-]*[$]?

       Usernames may only be up to 32 characters long.```

Reply:
I don't see how that will work. The entire purpose of this field was to establish safe and unique usernames ahead of time. If it doesn't do that, I don't see why we would add this field. We already have the first part of their email address as a proxy for this value.

We need these usernames to exist ahead of time so that the username in the SSH request already exists on the target system.

If we don't create these usernames ahead of time, how do we provision a Linux account for users before they SSH, and share that username with the SSH client so that it knows which username to use?